cpp/include/cudf/strings/find.hpp

```diff
@@ -140,6 +140,31 @@ std::unique_ptr<column> contains(
   rmm::cuda_stream_view stream      = cudf::get_default_stream(),
   rmm::device_async_resource_ref mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Returns a table of columns of boolean values for each string where true indicates
+ * the target string was found within that string in the provided column.
+ *
+ * Each column in the result table corresponds to the result for the target string at the same
+ * ordinal. i.e. 0th column is the boolean-column result for the 0th target string, 1th for 1th,
+ * etc.
+ *
+ * If the target is not found for a string, false is returned for that entry in the output column.
+ * If the target is an empty string, true is returned for all non-null entries in the output column.
+ *
+ * Any null string entries return corresponding null entries in the output columns.
+ *
+ * @param input Strings instance for this operation
+ * @param targets UTF-8 encoded strings to search for in each string in `input`
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @return New BOOL8 column
+ */
+std::unique_ptr<table> contains(
```

Comment:
One [suggestion](https://github.com/rapidsai/cudf/pull/15536#discussion_r1566415863) that @davidwendt had in #15536 was:
1. The function should take a `strings_column_view` instead of a `vector<string_scalar>`.
2. The function should return a LISTS column, where each list row has as many elements as the number of target strings.

I think the suggestion is for consistency with the other functions in the header.  But it will definitely complicate consumption of the results downstream.  We will need a way to split the lists column into N results columns for consumption.

Reply:
@davidwendt,  I think the first point is not intrusive, and should be doable.  I was wondering if you might be open to our leaving the return type a table instead of a lists column.

Reply:
Yes, the return type as a table should be ok.